File: 2800-minimum-string-length-after-removing-substrings/2800-minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        while True:
          patFound = False
          for i in range(len(s)-1):
            if s[i] == 'A' and s[i+1] == 'B':
              s = s[:i] + s[(i+2):]
              patFound = True 
              break
            
            if s[i] == 'C' and s[i+1] == 'D':
              s = s[:i] + s[(i+2):]
              patFound = True
              break
          
          if not patFound: 
            break

        return len(s)

            

File: 2800-minimum-string-length-after-removing-substrings/2800-minimum-string-length-after-removing-substrings.java
class Solution {
    public int minLength(String s) {
        Stack <Character> st = new Stack<>();

        for(char c : s.toCharArray()){
          if(st.isEmpty()){
            st.push(c);
            continue;
          }

          if(st.peek() == 'A' && c == 'B'){
            st.pop();
            continue;
          }

          if(st.peek() == 'C' && c == 'D'){
            st.pop();
            continue;
          }

          st.push(c);
        }

        return st.size();
    }
}